### PR TITLE
Fixes #38013 - Add Image mode card on host details tab

### PR DESCRIPTION
**Pull request by lfu**
#### What are the changes introduced in this pull request?
Add Image mode card on host details tab for bootc hosts

#### Considerations taken when implementing this change?
Depends on https://github.com/Katello/katello/pull/11209

#### What are the testing steps for this pull request?
Follow the test steps in https://github.com/Katello/katello/pull/11209.
Go to a bootc host - details tab.

### CODE REVIEW

**Comment by lfu** [comment]
image info

**Comment by jeremylenz** [comment]
I think it'd be cool to have our shiny new image mode icon on the card :)

**Comment by ianballou** [comment]
@sambible I think this card might be a good candidate to use as a stopping point for packit testing on what we've developed so far for image mode.

**Comment by lfu** [comment]
Not sure if it is shiny.
image icon

**Comment by jeremylenz** [comment]
Icon looks awesome :) but we maybe don't need so many! I was thinking just in the header. Maybe instead of "Image information" it could be "<icon> Image-mode host".

**Comment by lfu** [comment]
image mode

File: webpack/components/extensions/HostDetails/DetailsTabCards/ImageModeCard.js
Comment (ianballou) [inline webpack/components/extensions/HostDetails/DetailsTabCards/ImageModeCard.js:17]: "Image mode host" as a title more sounds like it's announcing that this is an image mode host rather than describing image mode host details.

What if we change it to:
```suggestion
      <span style={{ marginRight: '0.5rem' }}>{__('Image mode details')}</span>
```

File: webpack/components/extensions/HostDetails/DetailsTabCards/ImageModeCard.js
Comment (ianballou) [inline webpack/components/extensions/HostDetails/DetailsTabCards/ImageModeCard.js:36]: For any field that is empty, should we display a dash `-` like in the mock image?

image

**Review by ianballou — COMMENTED** [review]
Working great! Just a couple suggestions.

File: webpack/components/extensions/HostDetails/DetailsTabCards/ImageModeCard.js
Comment (ianballou) [reply webpack/components/extensions/HostDetails/DetailsTabCards/ImageModeCard.js:36]: This seems to work well:

```
diff --git a/webpack/components/extensions/HostDetails/DetailsTabCards/ImageModeCard.js b/webpack/components/extensions/HostDetails/DetailsTabCards/ImageModeCard.js
index 7ec258bf77..ce281d6274 100644
--- a/webpack/components/extensions/HostDetails/DetailsTabCards/ImageModeCard.js
+++ b/webpack/components/extensions/HostDetails/DetailsTabCards/ImageModeCard.js
@@ -19,6 +19,8 @@ const ImageModeCard = ({ isExpandedGlobal, hostDetails }) => {
     </>
   );
   if (!imageMode) return null;
+  const getValueOrDash = value => (value || '-');
+
   return (
     <CardTemplate
       header={header}
@@ -29,24 +31,24 @@ const ImageModeCard = ({ isExpandedGlobal, hostDetails }) => {
       <DescriptionList isHorizontal>
         <DescriptionListGroup>
           <Dt>{__('Running image')}</Dt>
-          <Dd>{hostDetails.content_facet_attributes.bootc_booted_image}</Dd>
+          <Dd>{getValueOrDash(hostDetails.content_facet_attributes.bootc_booted_image)}</Dd>
           <Dt>{__('Running image digest')}</Dt>
-          <Dd>{hostDetails.content_facet_attributes.bootc_booted_digest}</Dd>
+          <Dd>{getValueOrDash(hostDetails.content_facet_attributes.bootc_booted_digest)}</Dd>
 
           <Dt>{__('Staged image')}</Dt>
-          <Dd>{hostDetails.content_facet_attributes.bootc_staged_image}</Dd>
+          <Dd>{getValueOrDash(hostDetails.content_facet_attributes.bootc_staged_image)}</Dd>
           <Dt>{__('Staged image digest')}</Dt>
-          <Dd>{hostDetails.content_facet_attributes.bootc_staged_digest}</Dd>
+          <Dd>{getValueOrDash(hostDetails.content_facet_attributes.bootc_staged_digest)}</Dd>
 
           <Dt>{__('Available image')}</Dt>
-          <Dd>{hostDetails.content_facet_attributes.bootc_available_image}</Dd>
+          <Dd>{getValueOrDash(hostDetails.content_facet_attributes.bootc_available_image)}</Dd>
           <Dt>{__('Available image digest')}</Dt>
-          <Dd>{hostDetails.content_facet_attributes.bootc_available_digest}</Dd>
+          <Dd>{getValueOrDash(hostDetails.content_facet_attributes.bootc_available_digest)}</Dd>
 
           <Dt>{__('Rollback image')}</Dt>
-          <Dd>{hostDetails.content_facet_attributes.bootc_rollback_image}</Dd>
+          <Dd>{getValueOrDash(hostDetails.content_facet_attributes.bootc_rollback_image)}</Dd>
           <Dt>{__('Rollback image digest')}</Dt>
-          <Dd>{hostDetails.content_facet_attributes.bootc_rollback_digest}</Dd>
+          <Dd>{getValueOrDash(hostDetails.content_facet_attributes.bootc_rollback_digest)}</Dd>
         </DescriptionListGroup>
       </DescriptionList>
     </CardTemplate>
```

**Review by ianballou — COMMENTED** [review]
This is looking perfect to me.

I'd like to see if we can get a review from @sambible since we've merged a few imagae mode things without a check from a QE reviewer. I'll hold off on acking the PR for now.

**Comment by jeremylenz** [comment]
Looks great except I'd make the icon a bit bigger (and move it up vertically a bit so it's better aligned.)

File: webpack/components/extensions/HostDetails/DetailsTabCards/ImageModeCard.js
Comment (jeremylenz) [inline webpack/components/extensions/HostDetails/DetailsTabCards/ImageModeCard.js:20]: This could be moved outside the component so it doesn't get recreated on each render. Not essential here though, it shouldn't be too expensive :)

**Comment by ianballou** [comment]
> Looks great except I'd make the icon a bit bigger (and move it up vertically a bit so it's better aligned.)

It renders like this on Firefox:
image

File: webpack/components/extensions/HostDetails/DetailsTabCards/ImageModeCard.js
Comment (jeremylenz) [inline webpack/components/extensions/HostDetails/DetailsTabCards/ImageModeCard.js:22]: It's not actually a hyphen ;)

```suggestion
  const getValueOrDash = value => (value || '—');
```

**Comment by jeremylenz** [comment]
> It renders like this on Firefox:

oh! Yeah, that's much better..

**Review by ianballou — COMMENTED** [review]
ACK from me!

I'd still like to wait to merge until @sambible has time to test.

**Review by ianballou — APPROVED** [review]

**Comment by ianballou** [comment]
@lfu since it's been a while, I think we should just merge the PR. Feel free to merge when you're ready.

**Pull request merged**
